### ISSUE TITLE
Supplying file descriptions with &chkDesc does not work (except for the first file)

### ISSUE THREAD

**Issue by kriro** [Jako/FileDownloadR#5]
Supplying descriptions in a descriptions chunk does not work except for the first file. The reason is a wrong usage of ```array_walker```, which in its current implementation is effectively a no-op and hence does not actually trim the line breaks before string comparison. The first occurrence is here:

https://github.com/Jako/FileDownloadR/blob/dc0456d5a29777c04144dac0158886dee8e045b0/core/components/filedownloadr/src/FileDownloadR.php#L833

Instead of

```
        array_walk($linesX, function ($val) {
            return trim($val);
        });
```

we probably want

```
        array_walk($linesX, function (&$val) {
            $val = trim($val);
        });
```

to actually trim the array elements in-place.

After I applied the above change(s), descriptions work again for all files.

There are several instances throughout the code in ```FileDownloadR.php``` with the same issue, and it looks like this problem may affect other options, as well.

**Comment by Jako** [comment]
Can you check this fix, please!

**Comment by Jako** [comment]
Fixed with 3.1.3
